packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -165,6 +168,40 @@ export function listenToEvent(
   }
 }
 
+const validFBLegacyPrimerRels = new Set([
+  'dialog',
+  'dialog-post',
+  'async',
+  'async-post',
+  'theater',
+  'toggle',
+]);
+
+function willDeferLaterForFBLegacyPrimer(nativeEvent: any): boolean {
+  let node = nativeEvent.target;
+
+  while (node !== null) {
+    // Primer works by intercepting a click event on an <a> element
+    // that has a "rel" attribute that matches one of the valid ones
+    // in the Set above. If we intercept this before Primer does, we
+    // will need to defer the current event till later and discontinue
+    // execution of the current event. To do this we can add a document
+    // event listener and continue again later after propagation.
+    if (node.tagName === 'A' && validFBLegacyPrimerRels.has(node.rel)) {
```

Comment:
will this traversal be run on every event? if so, is it cheap to do?

Reply:
Only for click events.